starter/src/main/java/com/example/starter/base/impl/MainLayout.java

```diff
@@ -72,29 +66,12 @@ private void addNavigationItem(RouteData data) {
     }
 
     private String getCaption(RouteData data) {
-        try {
-            Bundle bundle = FrameworkUtil.getBundle(MainLayout.class);
-            ServiceReference<?>[] references = bundle.getBundleContext()
-                    .getAllServiceReferences(
-                            NavigationCaptionProvider.class.getName(), null);
-            Optional<NavigationCaptionProvider> routeCaptionProvider = Stream
-                    .of(references).map(bundle.getBundleContext()::getService)
-                    .map(NavigationCaptionProvider.class::cast)
-                    .filter(provider -> provider
-                            .isApplicable(data.getNavigationTarget()))
-                    .findFirst();
-            String routeCaption;
-            if (routeCaptionProvider.isPresent()) {
-                routeCaption = routeCaptionProvider.get()
-                        .getCaption(data.getNavigationTarget());
-            } else {
-                routeCaption = data.getTemplate();
-            }
-            return routeCaption;
-        } catch (InvalidSyntaxException e) {
-            // this may not happen with null as a filter
-            assert false;
-            return "";
+        Class<? extends Component> navigationTarget = data.getNavigationTarget();
+        PageTitle title = navigationTarget.getAnnotation(PageTitle.class);
+        if(title != null) {
+            return title.value();
+        } else {
+            return navigationTarget.getSimpleName();
```

Comment:
This is worse than previously used `data.getTemplate()` but since this is just an example where everything is implemented correctly it's not a big deal.